app/src/main/java/com/github/dedis/student20_pop/model/Event.java

```diff
@@ -0,0 +1,136 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Class modeling an Event
+ */
+public class Event {
+
+    private String name;
+    private long time;
+    private String id;
+    private String lao;
+    private List<String> attendees;
+    // Can use GeoLocation in the future
+    private String location;
+    // Can use enums in the future
+    private String type;
+    private JSONObject other;
+    private List<String> attestation;
+
+    /**
+     * Constructor for an Event
+     *
+     * @param name the name of the event, can be empty
+     * @param time the creation time, can't be modified
+     * @param lao the public key of the associated LAO
+     * @throws IllegalArgumentException if any of the parameters is null
+     */
+    public Event(String name, Date time, String lao, String location, String type) {
+        if(name == null || time == null || lao == null || location == null || type == null) {
+            throw new IllegalArgumentException("Trying to create an event with null parameters");
+        }
+        this.name = name;
+        this.time = time.getTime() / 1000L;
+        // Simple for now, will hash in the future
+        this.id = name + time;
+        this.lao = lao;
+        this.attendees = new ArrayList<>();
+        this.location = location;
+        this.type = type;
+        // Will get the list of witnesses, hash and sign in the future
+        this.attestation = new ArrayList<>(Collections.singletonList(name + time + lao + location));
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    /**
+     *
+     * @return creation time of the LAO as Unix Timestamp, can't be modified
+     */
+    public long getTime() {
+        return time;
+    }
+
+    /**
+     *
+     * @return ID of the event, can't be modified
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     *
+     * @return ID of the associated LAO
+     */
+    public String getLao() {
+        return lao;
+    }
+
+    /**
+     *
+     * @return list of public keys of the attendees
+     */
+    public List<String> getAttendees() {
+        return attendees;
+    }
+
+    public String getLocation() {
+        return location;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    /**
+     *
+     * @return list of signatures by the organizer and the witnesses of the corresponding LAO
+     */
+    public List<String> getAttestation() {
+        return attestation;
+    }
+
+    /**
+     *
+     * @param attendees list of public keys of attendees, can be empty
+     * @throws IllegalArgumentException if the list is null or at least one public key is null
+     */
+    public void setAttendees(List<String> attendees) {
+        if(attendees == null || attendees.contains(null)) {
+            throw new IllegalArgumentException("Trying to add a null attendee to the event " + name);
+        }
+        this.attendees = attendees;
+    }
```

Comment:
Is this really needed ? Aren't events one time objects with a fixed set of attendees at the creation ?

Reply:
The way I understood it, there's a possibility to create future events which don't have an attendees list/set yet